tests/fixtures/directory/project_with_transitive_directory_dependencies_missing/pyproject.toml

```diff
@@ -0,0 +1,12 @@
+[tool.poetry]
+name = "project-with-transitive-directory-dependencies-missing"
+version = "1.2.3"
+description = "This is a description"
+authors = ["Your Name <you@example.com>"]
+license = "MIT"
+
+[tool.poetry.dependencies]
+python = "*"
+foo = {path = "../foo/"}
```

Comment:
Notice that `foo` does not exist, I created it locally locked this project and then deleted it.

Reply:
Is it possible that this issue is not at all about transitive dependencies but just any dependencies?  If I don't miss anything this new fixture does not contain a transitive path dependency, but just a normal path dependency.

Further, it seems the issue can be reproduced with the existing `missing_directory_dependency` fixture. A test in `test_install.py` as follows should be sufficient, shouldn't it?

```
@pytest.mark.parametrize("options", ["", "--no-directory"])
def test_install_missing_directory_dependency_with_no_directory(
    command_tester_factory: CommandTesterFactory,
    project_factory: ProjectFactory,
    fixture_dir: FixtureDirGetter,
    options: str,
) -> None:
    poetry = _project_factory(
        "missing_directory_dependency", project_factory, fixture_dir
    )
    assert isinstance(poetry.locker, TestLocker)
    poetry.locker.locked(True)
    tester = command_tester_factory("install", poetry=poetry)
    if options:
        tester.execute(options)
    else:
        with pytest.raises(ValueError, match="does not exist"):
            tester.execute(options)
```

Then, we really don't need a new test in `test_lock.py`

Reply:
Well, it ends up being a transitive directory dependency because of how it's used in the test, but yes I like your test more 😄

Reply:
Strictly speaking, only keep the non-test part of the first one. 😉

Reply:
Yes I got you. I just did a rebase / squash that should do just that.